File: data/andrewshitov/Kotlin/Challenge89Part1.kt
fun gcd(a: Int, b: Int): Int {
    var x = a
    var y = b

    while (y != 0) {
        val t = y
        y = x % y
        x = t
    }

    return x
}

fun main(args: Array) {
    var n: Int
    if (args.size == 1)
        n = args[0].toInt()
    else
        n = 3

    var s = 0
    for (x in 1..n) {
        for (y in x + 1 .. n) {
            s += gcd(x, y)
        }
    }

    println(s)
}

File: data/andrewshitov/Scala/Challenge89Part1.scala
import math.BigInt

object Main {
    def main(args: Array[String]): Unit = {
        var n: Int = if (args.size == 1) args(0).toInt else 3

        var s: BigInt = 0
        for (x <- 1 to n) {
            for (y <- x + 1 to n) {
                s = s + BigInt(x).gcd(y)
            }
        }

        println(s)
    }
}